core/src/main/java/com/redhat/lightblue/client/response/DefaultLightblueDiagnosticsResponse.java

```diff
@@ -61,27 +63,44 @@ public boolean hasDiagnostics(String diagnosticsElementName) {
     public List<DiagnosticsElement> getDiagnostics() {
         List<DiagnosticsElement> diagnosticsElements = new ArrayList<>();
 
-        Iterator<String> fieldNames = getJson().fieldNames();
-        boolean isHealthy = false;
-        String message = null;
+        if (getJson() != null) {
+            Iterator<String> fieldNames = getJson().fieldNames();
+            boolean isHealthy = false;
+            String message = null;
```

Comment:
Still a bit weird that these are declared outside of the `while` block below. They are only needed within that block, so being declared in an outer scope makes it look like they are used in the outer scope when they aren't.

Not a huge deal.